frontend/src/routes.js

```diff
@@ -145,6 +146,7 @@ const routes = (
            element={<PrivateRoute element={AddEditCharacteristic}/>}/>
     <Route path={`${process.env.PUBLIC_URL}/characteristic/:uri/:viewMode`}
            element={<PrivateRoute element={AddEditCharacteristic}/>}/>
+    <Route path={`${process.env.PUBLIC_URL}/characteristic/:uri/view`} element={<PrivateRoute element={Characteristic}/>}/>
```

Comment:
this line should be on top of 148, since it is never gonna reach this line of code if you put this below the line 148. Keep in mind that :viewMode is a parameter